ovs/lib/storagedriver.py

```diff
@@ -165,71 +187,95 @@ def manual_voldrv_arakoon_checkup():
             storagerouter.invalidate_dynamics(['partition_config'])
             if len(storagerouter.partition_config[DiskPartition.ROLES.DB]) > 0:
                 available_storagerouters[storagerouter] = DiskPartition(storagerouter.partition_config[DiskPartition.ROLES.DB][0])
             all_sr_ips.append(storagerouter.ip)
-        if create_cluster is True and len(current_services) == 0 and len(available_storagerouters) > 0:
-            storagerouter, partition = available_storagerouters.items()[0]
-            result = ArakoonInstaller.create_cluster(cluster_name=cluster_name,
-                                                     ip=storagerouter.ip,
-                                                     base_dir=partition.folder)
-            current_services.append(add_service(storagerouter, result))
-            ArakoonInstaller.restart_cluster_add(cluster_name, current_ips, storagerouter.ip)
-            current_ips.append(storagerouter.ip)
-            StorageDriverController._configure_arakoon_to_volumedriver()
-
-        if 0 < len(current_services) < len(available_storagerouters):
+
+        if create_cluster is True and len(claimed_clusters) == 0 and len(current_services) == 0:  # Create new cluster
+            metadata = ArakoonInstaller.get_arakoon_metadata_by_cluster_type(cluster_type=ServiceType.ARAKOON_CLUSTER_TYPES.SD, in_use=False)
+            if len(metadata) == 0:  # No externally managed cluster found, we create 1 ourselves
+                if not available_storagerouters:
+                    raise RuntimeError('Could not find any Storage Router with a DB role')
+
+                storagerouter, partition = available_storagerouters.items()[0]
+                cluster_name = 'voldrv'
+                result = ArakoonInstaller.create_cluster(cluster_name=cluster_name,
+                                                         cluster_type=ServiceType.ARAKOON_CLUSTER_TYPES.SD,
+                                                         ip=storagerouter.ip,
+                                                         base_dir=partition.folder)
+                ports = [result['client_port'], result['messaging_port']]
+                ArakoonInstaller.restart_cluster_add(cluster_name=cluster_name, current_ips=current_ips, new_ip=storagerouter.ip)
+                current_ips.append(storagerouter.ip)
+                metadata = ArakoonInstaller.get_arakoon_metadata_by_cluster_type(cluster_type=ServiceType.ARAKOON_CLUSTER_TYPES.SD, in_use=False)
+            else:
+                ports = []
+                storagerouter = None
+
+            if len(metadata) == 0:
+                raise ValueError('No arakoon cluster found of type {0}'.format(ServiceType.ARAKOON_CLUSTER_TYPES.SD))
+
+            cluster_name = metadata[0].cluster_id
+            logger.info('Claiming {0} managed arakoon cluster: {1}'.format('externally' if storagerouter is None else 'internally', cluster_name))
+            metadata[0].claim()
```

Comment:
We should have locking here. This cluster might already be claimed by another codepath.